### PR TITLE
Add missing smarty variable

### LINKED ISSUE #27028: BO - Dashboard - errors are displayed when debug mode is enabled
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

After fresh install with develop branch, in the dashboard page
errors are displayed
image

### Expected behavior

_No response_

### Steps to reproduce

1. Fresh install develop branch
2. Go to BO > Dashboard page & debug mode enabled
3. See error

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.2, 7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Since #27015 has been merged, a variable still needed by some dashboard modules was removed leading to an exception when displaying the dashboard. This PR sets the needed variable back.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27028
| How to test?      | Please see #27028
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27057)

### CODE REVIEW

**Comment by atomiix** [comment]
I guess this PR can be considered as a temporary fix. The real solution would be to update the modules that still require the smarty variable previously removed.

File: classes/module/Module.php [hunk 1]
```diff
@@ -2182,6 +2182,7 @@ public function display($file, $template, $cache_id = null, $compile_id = null)
             $this->smarty->assign([
                 'module_dir' => __PS_BASE_URI__ . 'modules/' . basename($file, '.php') . '/',
                 'module_template_dir' => ($overloaded ? _THEME_DIR_ : __PS_BASE_URI__) . 'modules/' . basename($file, '.php') . '/',
+                'allow_push' => false,
```
Comment (eternoendless) [inline classes/module/Module.php]:
```suggestion
                'allow_push' => false, // required by dashboard modules
```

**Review by matks — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @atomiix

**Comment by matks** [comment]
Close and reopen to restart CI following merge of https://github.com/PrestaShop/PrestaShop/pull/27067

**Comment by matks** [comment]
Thank you @atomiix

**Comment by PrestaEdit** [comment]
You don't know how high I dislike and disagree with this kind of Pull Request ! 😕

**Comment by matks** [comment]
@PrestaEdit Instead of saying how much you are unhappy about this PR, why not rather explain the reasons behind?

To err is human, if we did a mistake by introducing this PR we'll be happy to hear about your explanations and maybe find a better way to fix https://github.com/PrestaShop/PrestaShop/issues/27028

So please leave the rant aside. We're listening, what's wrong with this PR?

**Comment by PrestaEdit** [comment]
The better way is not to re-introduce a pseudo backward about a develop branch because existing modules are not up-to-date.

There is surely some modules that will be no compliant with all the deprecated methods removed.
The best way is to update theses modules, according to the core, simply.

(And sorry, I was on vacation into Disnleyland Paris last two days and so I keep my comments simple as possible to be understandable this day :p)

**Comment by atomiix** [comment]
@PrestaEdit I'm sorry, I just saw that you started to work on dash* modules to remove push-related code, I did it too 😔

**Comment by PrestaEdit** [comment]
No worries, I understand what and why you do it. It's just that if we do that, we can't simply remove nothing as module that are no up-to-date will surely use something remved (as the push, here).

**Pull request merged**